src/openforms/prefill/contrib/kvk/plugin.py

```diff
@@ -37,53 +34,55 @@ def get_prefill_values(
         self, submission: Submission, attributes: List[str]
     ) -> Dict[str, Any]:
         # check if submission was logged in with the identifier we're interested
-        if not self.get_submission_attr(submission):
+        if not submission.kvk:
             return {}
 
-        client = KVKClient()
+        client = KVKProfileClient()
 
         try:
-            results = client.query(
-                **{self.get_query_param(): self.get_submission_attr(submission)}
-            )
+            result = client.query(submission.kvk)
         except (RequestException, ClientError, KVKClientError):
             return {}
 
-        data = self.select_item(results["resultaten"])
-        if not data:
-            return {}
+        self.modify_result(result)
 
         values = dict()
         for attr in attributes:
             try:
-                values[attr] = glom(data, attr)
+                values[attr] = glom(result, attr)
             except GlomError:
                 logger.warning(
                     f"missing expected attribute '{attr}' in backend response"
                 )
         return values
 
-    def select_item(self, items):
+    @classmethod
+    def modify_result(cls, result):
+        # move the desired item from the unordered list to a know place
+        address = cls.select_address(result["adressen"], "bezoekadres")
+        if address:
+            result["bezoekadres"] = address
+        address = cls.select_address(result["adressen"], "correspondentieadres")
+        if address:
+            result["correspondentieadres"] = address
+
+    @classmethod
+    def select_address(cls, items, type_):
```

Comment:
Does not have to be a classmethod, `cls` is unused. Unless this is supposed to be public API, I'd just move this into a plain helper function